packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1987,9 +1994,30 @@ export class ContainerRuntime
 		assert(this.outbox.isEmpty, 0x3cf /* reentrancy */);
 	}
 
-	public orderSequentially<T>(callback: () => T): T {
+	public orderSequentially<T>(callback: () => T, flush?: Deferred<void>): T {
```

Comment:
We cannot use a boolean here to enable/disabling this flush logic. If the code inside `orderSequentially` is async and we're using a boolean flag, the flag will be set to false at the end of this function, while the async code will continue running and thus schedule more flush calls from the runtime.